### DISCUSSION TITLE
TuringCourses/major_module/applied_operations_research/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#41] [General]
# TuringCourses/major_module/applied_operations_research/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major_module/applied_operations_research/

**Comment by chiakicage** [comment]
注意到对于本文作者来说4.5已经是往下拉均绩，我倾向于这课给分还算好（
课程本身是很有意思的，作业量非常少（一般一两个题）所以推荐耐心思考独立完成。期末（如果有的话）倾向于考证明题，并没有出手算单纯形的题（事实上单纯形算法的细节相当多）。论文阅读是个对个人能力提升很大的事情，有种对理论计算机科学“祛魅”的效果，如果能体会到乐趣的话这门课的体验是非常好的。

**Comment by NeilAlfred** [comment]
注意到对于本文作者来说4.5已经是往下拉均绩，并且在我拿了5.0的情况下，我倾向于这课的给分非常好（x

和chiakicage和ZhouTimeMachine一起组队完成了这门课。平时上课讲的内容难度有点离散，因为大部分偏向基础，但有些（少数）例子涉及到非常夸张的技巧和构造；基本上有 3 / 5 的课能够当堂听懂全部，剩下的需要课后补一下 PPT ，但最难的例子听不懂也没关系，作业题只需要把概念和定义听懂了就都能做。

平时的作业题基本难度不大，都是套用上课讲过的证明思路就能做出来；偶尔会遇到一两道需要自创额外证法的题目（基本集中在次模优化板块），大家讨论一下也能做出来。

张国川老师是一个极具讲课魅力的老师，鼓励大家开冲（x 另外，如果上课你能跟名为 zgc 的 NPC 成功互动，他会认为你的水平不错，平时成绩的印象分部分几乎会直接拉满（x

**Comment by Yuanrx1** [comment]
“如果数学对你来说没有那么讨厌，那么这门课值得一选”读到这句话的时候就已经猜到了这份课程攻略出自谁手，也让我有了一种强烈的想在评论区回复的愿望。请不要自嘲你的“理想主义”，有情众生皆为情怀，但我信总有人的情怀更为炽热、更令人感动。一生与真理相伴，这样的人一定和蔼而且浪漫——这是我心中的张老师，我猜未来的你也是这样。选择了这门课，我的大学不再有遗憾，轮到我庆幸。世界纷扰，你却写最安静的文字，请一直写下去。

**Comment by HobbitQia** [comment]
> 我们这是在纯粹的讨论学术问题，干干净净的，不要把那些乱七八糟的东西带到这里来。我们这只能讲真话，做真事，乌七八糟的事情出了教室去干，不要在这干。——张国川

如果计院的课里面一定要选一门神课的话，那我的答案只会是这门课。就像张老师说的，这里是最纯粹的课堂，老师、助教、学生每周来到这里只为了交流知识、讨论问题。老师的讲课非常引人入胜，穿插风趣的故事，还有他自己对学术界（尤其是机器学习领域）、社会等方面的尖锐评价。能感受到老师有文人的清高和恃才放旷，也非常希望大家未来能真正领导社会。

关于到课这件事，张老师如是说道：“你努力是你自己的本分，不是给我的馈赠。我教书是为了听课的学生，你读书只是为你自己。想清楚了就抓紧时间（退课）。”

除此之外这门课的考试非常非常难，可以在 98 上找到往年的回忆卷（然而毫无作用）笔者最后的期末考试没有发挥太好，因此最后的结果也不尽如人意。

> 我就说这课挺难的，大家还不信。这课难在哪呢，难就难在我还觉得很容易。——张国川

相比于成绩本身，这门课带给我的知识是值得的。在浙江大学声名远扬的计算机学院，大部分老师的课堂可以说是不堪回首，这里面也不乏查老师上的“暗黑天王”，我都快忘了老师传道授业解惑的样子。张老师依然坚守在这里，讲最纯粹的学术，守最原始的课堂，传最深刻的道理。

**Comment by Unc1eW4ng** [comment]
22级pre又成为了评分的一部分，本人和队友选择了某经典问题的前沿成果，由于某一个具体方向的运筹学论文通常会用同一套证明的框架，所以论文的作者对框架性的内容都介绍的比较少，而然这又是理解整篇论文的核心所在，导致阅读起来难以下手，lz和队友选择递归式地阅读之前的成果，终于找到了某篇比较合适的落脚点，理解完框架之后就相对好做一些。这个pre的另外的难度来自于上台讲的时候，张老师和组里的大牛（甚至是你准备的这篇论文的一作）亲临拷打，临场的压力会比较巨大。本人在准备的时候已经想了很多张老师可能会拷打的地方，最后讲的时候还是会被问的措手不及。虽然会有批评，不过不必担心，张老师也承认pre运筹学的论文难度比较大，能体会到同学们的用心。
